app/src/main/java/com/sun/moviedb_45/data/source/MovieRepository.java

```diff
@@ -0,0 +1,32 @@
+package com.sun.moviedb_45.data.source;
+
+import com.sun.moviedb_45.data.source.local.MovieLocalDataSource;
+import com.sun.moviedb_45.data.source.remote.MovieRemoteDataSource;
+import com.sun.moviedb_45.service.response.MovieResponse;
+
+import io.reactivex.Observable;
+
+public class MovieRepository implements MovieDataSource.Local, MovieDataSource.Remote {
+    private static MovieRepository sInstance;
+    private MovieRemoteDataSource mRemoteDataSource;
+    private MovieLocalDataSource mLocalDataSource;
```

Comment:
```suggestion
    private MovieDataSource.Local mLocalDataSource;
```
the high-level module must never rely on any low-level module but should depend on abstractions